components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -2311,6 +2345,26 @@ public List<IdPGroup> getValidIdPGroupsByIdPGroupIds(List<String> idpGroupIds, S
         }
     }
 
+    @Override
+    public FederatedAuthenticatorConfig[] getAllFederatedAuthenticators(String tenantDomain)
+            throws IdentityProviderManagementException {
+
+        List<FederatedAuthenticatorConfig> userDefinedAuthenticators =
```

Comment:
This list is not userDefinedAuthenticators eventually. You seems to be populating that will all authenticators. Either define the list and populate from each authenticator list or change the name